include/cuco/static_set_ref.cuh

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuco/detail/equal_wrapper.cuh>
+#include <cuco/function.hpp>
+#include <cuco/sentinel.cuh>  // TODO .hpp
+
+#include <cuda/std/atomic>
+
+namespace cuco {
+namespace experimental {
+
+/**
+ * @brief Device reference of static_set.
+ */
+template <typename Key,
+          cuda::thread_scope Scope,
+          typename KeyEqual,
+          typename ProbingScheme,
+          typename StorageRef,
+          typename... Functions>
+class static_set_ref
+  : public detail::function_impl<
+      Functions,
+      static_set_ref<Key, Scope, KeyEqual, ProbingScheme, StorageRef, Functions...>>... {
+ public:
+  using key_type            = Key;                             ///< Key Type
+  using probing_scheme_type = ProbingScheme;                   ///< Type of probing scheme
+  using storage_ref_type    = StorageRef;                      ///< Type of slot storage ref
+  using window_type = typename storage_ref_type::window_type;  ///< Probing scheme element type
+  using value_type  = typename storage_ref_type::value_type;   ///< Probing scheme element type
+  using size_type   = typename storage_ref_type::size_type;    ///< Probing scheme size type
+  using key_equal =
+    detail::equal_wrapper<value_type, KeyEqual>;  ///< Type of key equality binary callable
```

Comment:
I don't think this is right. The `key_equal` typedef should just be `KeyEqual`. However, the data member stored for `key_equal_` can be `equal_wrapper<value_type, KeyEqual>`.

Reply:
Good point.